appenders/python/pulse/handlers.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import atexit
+from logging.handlers import MemoryHandler
+
+from pulse.batcher import PulseBatcher
+
+
+# noinspection PyPep8Naming
+class PulseHandler(PulseBatcher, MemoryHandler):
+    """
+    A class which sends records to the Pulse log collector. Note that records
+    are converted to JSON upon posting. You must provide a
+    :class:`logging.Formatter` that returns the record as a dict.
+    """
+    # Note: Method Resolution Order (MRO):
+    #       self -> PulseBatcher -> MemoryHandler -> object
+    # Some methods in this class override methods in the order noted above
+
+    def __init__(self, endpoint, capacity=1000, flushLevel=logging.ERROR,
+                 threadCount=1, logger=None):
+        """
+        Initializes PulseHandler using the REST API endpoint, buffer capacity,
+        buffer flush level and thread count.
+
+        :param endpoint: The REST API endpoint for the Pulse Log Collector
+        :type endpoint: str
+        :param capacity: Number of records to buffer before flushing.
+                         Defaults to 1000.
+        :type capacity: int
+        :param flushLevel: Log level at which to to flush the buffer.
+        :type flushLevel: Log Level
+        :param threadCount: Number of threads to handle post requests.
+        :type threadCount: int
+        :param logger: :class:`logging.Logger` object for debug logging. If none
+                       is provided, a StreamHandler will be created to log to
+                       sys.stdout. It is recommended that you provide a
+                       logger if you plan to use more than one instance of this
+                       class.
+        :type logger: logging.Logger
+        :rtype: PulseHandler
+        """
+        PulseBatcher.__init__(self, endpoint, capacity, threadCount, logger)
+        MemoryHandler.__init__(self, capacity, flushLevel)
+        # Cleanup when Python terminates
+        atexit.register(self.close)
+
+    def setFormatter(self, fmt):
+        """
+        Set the formatter for this handler.
+
+        :param fmt: Formatter to apply to log records.
+        :type fmt: logging.Formatter
+        """
+        # Check if formatter provided returns a dictionary
+        #
+        # Note: Overrides MemoryHandler.setFormatter
```

Comment:
Should this comment go in the method comment block?

Reply:
Or switch with "Check if formatter provided returns a dictionary"

Reply:
Switched them around